spec/rspec/core/did_you_mean_spec.rb

```diff
@@ -0,0 +1,41 @@
+require 'tmpdir'
+require 'rspec/support/spec/in_sub_process'
+
+module RSpec
+  module Core
+    RSpec.describe DidYouMean do
+      if defined?(::DidYouMean::SpellChecker)
+        describe 'Call to DidYouMean' do
```

Comment:
We typically do `describe "#method_name"` for the outer, so `#call` in this case, with inner contexts so ideally this spec would have the layout, you can also use `hide_const` to test the functionality.
```
describe "#call" do
  if defined?(::DidYouMean::SpellChecker)
  context "when `DidYouMean::SpellChecker` is available", skip: defined?(::DidYouMean::SpellChecker) do
    # ...
  end
  context "when `DidYouMean::SpellChecker` is not available" do
    before do
      hide_const("DidYouMean::SpellChecker") if defined?(DidYouMean::SpellChecker)
    end
    # ...
  end
end
```

Reply:
This is neat, I was not aware of the hide_const.  I have implemented this change, but had one puzzle. When I run ```rspec spec``` then I get following error:
```
 RSpec::Core::DidYouMean#call when `DidYouMean::SpellChecker` is not available Success returns a hint
     Failure/Error: checker = ::DidYouMean::SpellChecker.new(:dictionary => Dir["spec/**/*.rb"])
     
     NameError:
       uninitialized constant DidYouMean::SpellChecker
       Did you mean?  DidYouMean::SpellChecker
                      DidYouMean::SPELL_CHECKERS
     # ./lib/rspec/core/did_you_mean.rb:15:in `call'
     # ./spec/rspec/core/did_you_mean_spec.rb:34:in `block (5 levels) in <module:Core>'
```
but when I run ```script/run_build``` all rspec tests pass.  Is there some filter or setting I should be adding when I do ```rspec spec``` on my local machine?

Reply:
The CI tests have finished and it appears that the ```:skip``` keyword is not supported by the rspec that runs with the earlier rubies e.g. 1.8.7.  Any suggestions what to do about this?

Reply:
Thinking about the fact I get the error when running ```rspec spec``` but not when I run ```script/run_build```,  I think that although ruby is an interpreted language, there seems to be a two pass  process that can see that the `if defined?(::DidYouMean::SpellChecker)`  at did_you_mean:12,  can be decided ahead of time.  

The issue seems to be that when I run 'rspec spec' this is done after the `did_you_mean_spec.rb` code runs but before the `did_you_mean.rb` code runs. Whereas when I do `script/run_build`, the static compilation step is done prior to the `did_you_mean_spec.rb` code running. Does this make sense?